### ISSUE TITLE
docker-compose up failed, gnu-crypt.h missing

### ISSUE THREAD

**Issue by artonge** [puffinrocks/puffin#20]
Hi,

I've tried to start puffin on a raspberry pi, but I've got the following error during the build process of the puffin image:

```
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Isrc/crypt_blowfish-1.3 -I/usr/local/include/python3.6m -c build/temp.linux-armv7l-3.6/_bcrypt.c -o build/temp.linux-armv7l-3.6/build/temp.linux-armv7l-3.6/_bcrypt.o
    In file included from /usr/local/include/python3.6m/Python.h:39:0,
                     from build/temp.linux-armv7l-3.6/_bcrypt.c:28:
    src/crypt_blowfish-1.3/crypt.h:17:23: fatal error: gnu-crypt.h: No such file or directory
     #include <gnu-crypt.h>
                           ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-y5dgfqms/bcrypt/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-8i0pdy9l-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-y5dgfqms/bcrypt/
ERROR: Service 'puffin' failed to build: The command '/bin/sh -c pip install --no-cache-dir -r requirements.txt' returned a non-zero code: 1
```

I understand that `gnu-crypt.h` is not available in the puffin docker image. Could this be right ? Or should it be installed on the host ?

**Comment by loomchild** [comment]
Hi,

Sorry for late reply. 

That's strange. Perhaps try to update your base image (git pull python:3-onbuild) before building. Also you shouldn't need to build the image in order to use it, unless you need to modify it somehow - did you try to download it from dockerhub?

Perhaps check on RaspberryPi related forums, since it doesn't seem to be related to Puffin, but rather to Python on bcrypt on this platform. If you can't find a solution, I will try to take a look at it over the weekend.

**Comment by artonge** [comment]
Hi,

Apparently it's because of a security update ([stackoverflow](https://stackoverflow.com/questions/49627807/cryptacular-is-broken)). Moving to `3.6.4-onbuild` does the trick.

FYI: onbuild images seems to be [deprecated](https://hub.docker.com/_/python/), maybe puffin should move to another base image.

**Comment by loomchild** [comment]
Interesting, thanks for info. I will look into migrating from -onbuild when I find some time.

**Comment by loomchild** [comment]
I dropped -onbuild and fixed on 3.6.4 since I had the same issue when building an image from scratch. Thank you for contribution and sorry for delay.

**Comment by loomchild** [comment]
Even better - I upgraded `bcrypt` to 3.x and was able to switch to latest 3.6.5.
